homeassistant/components/husqvarna_automower_ble/config_flow.py

```diff
@@ -108,23 +118,15 @@ async def async_step_user(
             if not _pin_valid(user_input[CONF_PIN]):
                 errors["base"] = "invalid_pin"
             else:
-                self.address = user_input[CONF_ADDRESS]
+                self.address = format_mac(user_input[CONF_ADDRESS]).upper()
```

Comment:
Do we need to migrate existing config entries?
I don't think it's a great idea to call `upper()` here, the point of `format_mac` is to have a consistent formatting of MAC addresses.

Reply:
Well, I had the intention to ease typing in the address and avoid issues with the library, as I sometimes had issues connecting when the address is formatted differently (lower case). For displaying the Mac later on, that is formatted separately as far as I know.

Maybe it makes also sense to introduce a `config validator` for mac addresses?